--- /tmp/old/libfuzzer/FuzzerInternal.h	2022-11-15 22:17:49.000000000 -0500
+++ /tmp/new/libfuzzer/FuzzerInternal.h	2022-12-02 06:55:00.000000000 -0500
@@ -65,7 +65,7 @@
   static void StaticFileSizeExceedCallback();
   static void StaticGracefulExitCallback();
 
-  void ExecuteCallback(const uint8_t *Data, size_t Size);
+  bool ExecuteCallback(const uint8_t *Data, size_t Size);
   bool RunOne(const uint8_t *Data, size_t Size, bool MayDeleteFile = false,
               InputInfo *II = nullptr, bool ForceAddToCorpus = false,
               bool *FoundUniqFeatures = nullptr);
